internal/worker/charmrevision/worker.go

```diff
@@ -66,63 +138,528 @@ func (config Config) Validate() error {
 	return nil
 }
 
+type revisionUpdateWorker struct {
```

Comment:
This whole worker should be called `charmrefresher` (for instance) instead of `charmrevision` for two reasons:
- it does more than just update its revision
- `refresher` is a verb like most of our workers

Reply:
I agree; I think we can settle on `charmrevisioner`.